java/flight/src/main/java/org/apache/arrow/flight/FlightClient.java

```diff
@@ -77,11 +81,21 @@
   /**
    * Create a Flight client from an allocator and a gRPC channel.
    */
-  private FlightClient(BufferAllocator incomingAllocator, ManagedChannel channel) {
+  private FlightClient(BufferAllocator incomingAllocator, ManagedChannel channel,
+      List<FlightClientMiddleware.Factory> middleware) {
     this.allocator = incomingAllocator.newChildAllocator("flight-client", 0, Long.MAX_VALUE);
     this.channel = channel;
-    blockingStub = FlightServiceGrpc.newBlockingStub(channel).withInterceptors(authInterceptor);
-    asyncStub = FlightServiceGrpc.newStub(channel).withInterceptors(authInterceptor);
+    final ClientInterceptor[] interceptors;
+
+    if (middleware.isEmpty()) {
```

Comment:
is there a reason to specialize here (does having a do nothing ClientInterceptorAdaptor add appreciable overhead?)